src/engine/WordIndexScan.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <string>
+
+#include "./Operation.h"
+
```

Comment:
Add comments on what this class doesn, most notably over the class and its constructors.